packages/dropdown/src/react/__stories__/index.story.js

```diff
@@ -578,3 +580,167 @@ storiesOf('portal', module).add('position', _ => (
     ultrices eros in cursus turpis massa.
   </div>
 ))
+const DropdownWithIcon = forwardRef(
+  (
+    {
+      appearance,
+      disabled,
+      className,
+      error,
+      label,
+      menu,
+      onChange,
+      onClick,
+      placeholder,
+      size,
+      subLabel,
+      style,
+      value,
+      icon,
+      ...rest
+    },
+    forwardedRef
+  ) => {
+    const {
+      combinedRef,
+      handleKeyDown,
+      handleMenuChange,
+      handleToggleOpen,
+      inNode,
+      isOpen,
+      longestMenuItemState,
+      menuPosition,
+      menuRef,
+      selectedLabel,
+      selectedValue,
+      setMenuPosition,
+      setOpen,
+      width
+    } = useDropdown({
+      forwardedRef,
+      menu,
+      value,
+      onChange,
+      onClick,
+      placeholder
+    })
+    return (
+      <Dropdown.Layout
+        disabled={disabled}
+        style={style}
+        className={className}
+        onKeyDown={handleKeyDown}
+        label={<Dropdown.Label label={label} />}
+        menu={
+          <DropdownContext.Provider value={selectedValue}>
+            <Dropdown.Menu
+              inNode={inNode}
+              isOpen={isOpen}
+              menu={menu}
+              menuPosition={menuPosition}
+              onClick={handleMenuChange}
+              onClose={() => {
+                setOpen(false)
+              }}
+              ref={menuRef}
+              width={width}
+            />
+          </DropdownContext.Provider>
+        }
+        subLabel={<Dropdown.SubLabel subLabel={subLabel} />}
+        button={
+          <Dropdown.Button
+            appearance={appearance}
+            disabled={disabled}
+            error={error}
+            isOpen={isOpen}
+            onClick={handleToggleOpen}
+            setMenuPosition={setMenuPosition}
+            size={size}
+            ref={combinedRef}
+            {...rest}
+          >
+            {icon}
+            <div style={{ height: '100%', position: 'relative', flex: 1 }}>
+              <Dropdown.Selected
+                appearance={appearance}
+                label={longestMenuItemState.label}
+                placeholder={placeholder}
+                selectedLabel={selectedLabel}
+                size={size}
+              />
+            </div>
+          </Dropdown.Button>
+        }
+      />
```

Comment:
If we wanted to not do this refactor we'd need to modify this part of the existing component structure to support icon and add the current `css/index.js` mods.